### PR TITLE
null safety in debezium to airbyte message generation

### LINKED ISSUE #41620: Debezium events missing a "before" node crashes source-postgres
### Connector Name

source-postgres

### Connector Version

3.4.24

### What step the error happened?

During the sync

### Relevant information

The code in RelationalDbDebeziumEventConverter is assuming that a change event read from debezium will have a "before" event.
It looks like the assumption is not always correct and causes a crash due to nullability.
We should make our code resilient to the case and understand when this missing "before" can happen

### Relevant log output

```shell
Seen at a [customer workspace](https://cloud.airbyte.com/workspaces/5473ee37-2fa9-4659-81fc-a181840ed152/connections/c1779998-f090-41f6-bc02-24cde17c1c09/job-history#14264450::4)
```

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by rodireich**
## What
Fixes https://github.com/airbytehq/airbyte/issues/41620

Fix null safety in java to kotlin code causing debezium event processing error if a "before" node is missing on the change event.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by theyueli — APPROVED** [review]
Thanks for the quick fix!

**Comment by rodireich** [comment]
I applied a dev build on the affected workspace which helped uncover the root cause.
(we should still put this fix in as null safety issue existed and masked the root cause here)

File: airbyte-integrations/connectors/source-postgres/build.gradle [hunk 1]
```diff
@@ -12,9 +12,9 @@ java {
 }
 
 airbyteJavaConnector {
-    cdkVersionRequired = '0.40.10'
+    cdkVersionRequired = '0.41.1'
     features = ['db-sources', 'datastore-postgres']
-    useLocalCdk = false
+    useLocalCdk = true
```
Comment (johnny-schmidt) [inline airbyte-integrations/connectors/source-postgres/build.gradle:17]: Was this intentional or done for testing?

Comment (rodireich) [reply airbyte-integrations/connectors/source-postgres/build.gradle:17]: Forgot to reply - sorry
This is the usual way of operation when making changes in CDK.
Until that CDK is published as a new version, the connector needs to consume a locally built CDK.
Once you publish a CDK we change this back to false ahead of merging.

**Pull request merged**